.ci/pitest.sh

```diff
@@ -172,6 +172,21 @@ pitest-utils)
   checkPitestReport "${ignoredItems[@]}"
   ;;
 
+pitest-JavaAstVisitor)
+  mvn -e -P$1 clean test org.pitest:pitest-maven:mutationCoverage;
+  declare -a ignoredItems=(
+  # The two following ignored items are to improve performance on deeply nested expressions, such as
+  # deeply concatenated Strings. AST construction would not fail if this code was removed
+  # under normal circumstances, but for deeply nested expressions, this optimization helps to avoid
+  # out of memory errors. Any test that would kill these mutations would take around 6 minutes or
+  # more to parse.
+  "JavaAstVisitor.java.html:<td class='covered'><pre><span  class='survived'>        if (binOpList.isEmpty()) {</span></pre></td></tr>"
+  "JavaAstVisitor.java.html:<td class='covered'><pre><span  class='survived'>        while (firstExpression instanceof CheckstyleJavaParser.BinOpContext) {</span></pre></td></tr>"
```

Comment:
If anyone has an idea of how to kill these mutations, please share.

Reply:
>> if (binOpList.isEmpty()) 

This means that both ways are equivalent. The silly version
```java
            final DetailAstImpl leftChild = visit(ctx.children.get(0));
            bop.addChild(leftChild);
```
work exactly the same way as the optimized version, but slowly.

You can try to add a test for something like
```java
String s = "" + "" + "" ...   a lot of ... + "";
```
to make non-optimized version fail with StackOverflowError.

Reply:
After some investigation, the time concern (6+ minutes) was really related to the printing of the AST; we can parse it in < 12 seconds.  Added test case in `JavaParserTest.java`, and both mutations are now killed.

Reply:
resolved.